### PR TITLE
cyclops cli: support for update command 

### LINKED ISSUE #454: support updating modules using cyctl
**Is your feature request related to a problem? Please describe.**
cyctl can currently create, list and delete modules. It would be useful to update modules using cyctl. For the start, cyctl should support updating the `spec.values`.

The command invocation should look like this:
```sh
cyctl update module <module-name> --key=<key> --value=<value>
```

If there is already a deployed module defined like this:
```yaml
apiVersion: cyclops-ui.com/v1alpha1
kind: Module
metadata:
  name: test
spec:
  template:
    path: path
    repo: repo
    version: version
  values:
    scaling:
      replicas: 2
      cpu: 1
    networking:
      expose: true
```
cyctl should support updating just the number of replicas with the command:
```sh
cyctl update module test --key="scaling.replicas" --value=3
```

### PR DESCRIPTION
**Pull request by ashish111333**
closes #454 

## 📑 Description
adds support for the updating modules using cyctl cli

## ✅ Checks
- [ ] I have updated the documentation as required
- [x] I have performed a self-review of my code

##  Additional context
cyclops  cli currently supports create,delete,describe,get ,this PR adds support for update

### CODE REVIEW

File: cyctl/internal/update/modules.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+var (
+	updateModuleExample = `# updates the module,takes [module-name] as an argument with flags --key and --value
+# to update replicas for a module named test 
+cyctl update module test --key="scaling.replicas" --value=3
+	`
+)
+
+// updates the given module from cyclops API
+// currently supports updating replica count for a module
```
Comment (petar-cvit) [inline cyctl/internal/update/modules.go:22]: This should support everything, not just replicas so you can remove this comment. Replicas were just an example

File: cyctl/internal/update/modules.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+func updateModule(clientset *client.CyclopsV1Alpha1Client, moduleName, key string, value int) {
```
Comment (petar-cvit) [inline cyctl/internal/update/modules.go:23]: Can you make the value type `interface`?

**Review by petar-cvit — CHANGES_REQUESTED** [review]
Thanks @ashish111333, left two comments

File: cyctl/internal/update/modules.go (hunk 2 above)
Comment (ashish111333) [reply cyctl/internal/update/modules.go]: done

File: cyctl/internal/update/modules.go (hunk 1 above)
Comment (ashish111333) [reply cyctl/internal/update/modules.go]: done

**Comment by ashish111333** [comment]
have done the requested changes,can you review it now @petar-cvit

**Review by petar-cvit — CHANGES_REQUESTED** [review]
Thanks @ashish111333, left a couple of comments

File: cyctl/internal/update/modules.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		return
+	}
+
+	module, err := clientset.Modules("cyclops").Get(moduleName)
+	if err != nil {
+		fmt.Println("Failed to fetch module ", err)
+		return
+
+	}
+	SpecValuesMap := make(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (ashish111333) [reply cyctl/internal/update/modules.go]: done

File: cyctl/internal/update/modules.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	err = json.Unmarshal(module.Spec.Values.Raw, &SpecValuesMap)
+	if err != nil {
+
+		fmt.Println("failed to decode json data:", err)
+		return
+
+	}
+	err = unstructured.SetNestedField(SpecValuesMap, value, key)
```
[the remark this answers is not in the document]
Comment (ashish111333) [reply cyctl/internal/update/modules.go]: done

File: cyctl/internal/update/modules.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	updatedModule, err := clientset.Modules("cyclops").Update(module)
+	if err != nil {
+		fmt.Println("failed to update module: ", err)
+		return
+	}
+	fmt.Printf("successfully updated %v", moduleName)
+}
+
+var (
+	UpdateModule = &cobra.Command{
```
Comment (petar-cvit) [inline cyctl/internal/update/modules.go:71]: Can you update the name to `updateModuleCMD`. Update command name in init func as well

**Review by petar-cvit — CHANGES_REQUESTED** [review]
@ashish111333 Build is failing because you need to unpack split output using `strings.Split(key, ".")...`. Also, remove unused `updatedModule` variable

File: cyctl/internal/update/modules.go (hunk 5 above)
Comment (ashish111333) [reply cyctl/internal/update/modules.go:71]: yeah

**Review by petar-cvit — APPROVED** [review]
Thanks @ashish111333 🧡

**Pull request merged**